app/src/main/java/org/simple/clinic/drugs/selection/custom/CustomDrugEntrySheet.kt

```diff
@@ -142,6 +143,10 @@ class CustomDrugEntrySheet : BaseBottomSheet<
     router.pop()
   }
 
+  override fun goToEditMedicineScreen() {
+    router.clearHistoryAndPush(PrescribedDrugsScreenKey(screenKey.patientUuid))
```

Comment:
Clearing history means it will drop all the screens before edit medicines screen as well. You can try one of `popUntil`, `popUntilInclusive` or `replaceKeyOfSameType`, see which one fits this use case.